tests/file_system_cache_test.py

```diff
@@ -122,5 +123,19 @@ def test_file_bigger_than_cache(self):
       cache.put("foo", b"bar")
       self.assertEqual(cache.get("foo"), None)
 
+  def test_threads(self):
+    def call_multiple_puts_and_gets(cache):
+      for i in range(100):
+        cache.put("foo", b"bar")
```

Comment:
"bar" is only 3 bytes, and it's most likely too small to be not atomic...so we won't be testing the limits of cache write.. we may need a cache write bigger than 64KB or PIPE_MAX size, see

https://stackoverflow.com/questions/7236475/what-happens-if-a-write-system-call-is-called-on-same-file-by-2-different-proces

also, since all the threads write "bar", it's hard to get corruption too...Maybe we should have a set of larger-than-64KB values that each thread can choose from to write, and then at the end of test, just hash contents from each cache read and make sure that the md5sum is one of the expected values